core/CMakeLists.txt

```diff
@@ -17,7 +17,7 @@ set(PRIVATE_LIBS
     bimg
 )
 
-add_library(knoting ${KNOTING_SOURCES} ${KNOTING_SHADERS})
+add_library(knoting ${KNOTING_SOURCES} ${KNOTING_SHADERS} src/engine.cpp inc/knoting/engine.h inc/knoting/subsystem.h)
```

Comment:
It is weird that CLion is adding these extra files here. Let's check tomorrow how it can be avoided, there must be an option somewhere.

Reply:
something to look into tomorrow for sure c: